opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -1212,6 +1213,344 @@ Dune::cpgrid::Intersection CpGrid::getParentIntersectionFromLgrBoundaryFace(cons
+
+
+    // Predict how many new cell ids per process are needed.
+    std::vector<std::size_t> cell_ids_needed_by_proc(comm().size());
+    std::size_t local_cell_ids_needed = 0;
+    for ( const auto& element : elements( levelGridView(0), Dune::Partitions::interior) ) {
+        // Get old mark (from level zero). After calling adapt, all marks are set to zero.
+        bool hasBeenMarked = currentData().front()->getMark(element) == 1;
+        if ( hasBeenMarked ) {
+            const auto& level = assignRefinedLevel[element.index()];
+            // Shift level (to level -1) since cells_per_dim_vec stores number of subdivisions in each direction (xyz)
+            // per parent cell, per level, starting from level 1, ..., maxLevel.
+            local_cell_ids_needed += cells_per_dim_vec[level-1][0]*cells_per_dim_vec[level-1][1]*cells_per_dim_vec[level-1][2];
+        }
+    }
+    comm().allgather(&local_cell_ids_needed, 1, cell_ids_needed_by_proc.data());
+
+    // Overestimate ('predict') how many new point ids per process are needed.
+    // Assign for all partition type points a 'candidate of global id' (unique in each process).
+    std::vector<std::size_t> point_ids_needed_by_proc(comm().size());
+    std::size_t local_point_ids_needed = 0;
+    for (std::size_t level = 1; level < cells_per_dim_vec.size()+1; ++level){
+        if(lgr_with_at_least_one_active_cell[level-1]>0) {
+            // Amount of local_point_ids_needed might be overestimated.
+            for (const auto& point : vertices(levelGridView(level))){
+                // If point coincides with an existing corner from level zero, then it does not need a new global id.
+                if ( !(*current_data_)[level]->corner_history_.empty() ) {
+                    const auto& bornLevel_bornIdx =  (*current_data_)[level]->corner_history_[point.index()];
+                    if (bornLevel_bornIdx[0] == -1)  { // Corner is new-> it needs a new global id
+                        local_point_ids_needed += 1;
+                    }
+                }
+            }
+        }
+    }
+    comm().allgather(&local_point_ids_needed, 1, point_ids_needed_by_proc.data());
+
+    auto expected_max_globalId_cell = std::accumulate(cell_ids_needed_by_proc.begin(),
+                                                      cell_ids_needed_by_proc.end(),
+                                                      max_globalId_levelZero + 1);
+    min_globalId_cell_in_proc = std::accumulate(cell_ids_needed_by_proc.begin(),
+                                                cell_ids_needed_by_proc.begin()+comm().rank(),
+                                                max_globalId_levelZero + 1);
+    min_globalId_point_in_proc = std::accumulate(point_ids_needed_by_proc.begin(),
+                                                 point_ids_needed_by_proc.begin()+ comm().rank(),
+                                                 expected_max_globalId_cell);
```

Comment:
I added the plus 1. I couldn't tell now if I forgot to add it, or if it's not necessary, but sure thing is that that does not hurt since, in some cases, we are overestimating anyway.